src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/ImmutableCollection.cs

```diff
@@ -0,0 +1,72 @@
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace CoreWCF.IdentityModel
+{
+
+    //
+    // A collection that can be made immutable by calling the
+    // MakeReadOnly method. Once the collection is made read-only
+    // Add, Remove and Clear methods will throw an exception 
+    // failing to add a item to the collection.
+    //
+    internal sealed class ImmutableCollection<T> : Collection<T>, IList<T>, IList
+    {
```

Comment:
Do we need this or can we add a reference to the nuget package System.Collections.Immutable? The difference between those data structures and this is this class can be switched from mutable to immutable. Replacing this with for example ImmutableList\<T\> would require everywhere that MakeReadOnly is called to construct an ImmutableList\<T\> from the mutable one and replace it. Thoughts?

Reply:
I doubled checked, we are not using anywhere... so deleting the file